docs/api-specs/reporting-api-v1.yaml

```
@@ -182,11 +215,167 @@ paths:
                 $ref: '#/components/schemas/ReportData'
         404:
           description: report data is not available yet (when either the calculation id is not valid or the calculation is still running)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
         401:
           description: If the access token does not grant permission to this Report
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+  # notification
+
+  /webhook:
+    get:
+      security:
+        - development:
+          - reporting_read
+      tags:
+        - notification
+      summary: Return list of existing Webhooks
+      responses:
+        200:
+          description: List of all available Webhooks the requester has created, empty array if no Webhooks are available
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Webhook'
+        401:
+          description: If no valid access token
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        403:
+          description: No permissions to access the resource
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+  /webhook/{webhookId}:
+    get:
+      security:
+        - development:
+          - reporting_read
+      tags:
+        - notification
+      summary: Return a specific Webhook
+      parameters:
+        - in: path
+          name: webhookId
+          schema:
+            type: string
+          required: true
+      responses:
+        200:
+          description: Get details of a webhook, including details of subscribed events and configuration
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Webhook'
+        401:
+          description: If no valid access token
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        403:
+          description: No permissions to access the resource
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        404:
+          description: Resource not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+  /webhook/{webhookId}/subscribe/report/{reportId}:
+    post:
+      security:
+        - development:
+          - reporting_write
+      tags:
+        - notification
+      summary: Subscribe to events for a specific Report
```

```suggestion
      summary: Subscribe to events for a specific Report
      description: If you subscribe to a reportId, Aam Digital will continuously monitor the changes users make in the application and check if they affect the results calculcated by that report. Whenever this is the case, a new report-calculation is automatically triggered for you and once the calculation is completed you receive an event containing that calculationId. You can use this to immediately fetch to available report data from the /report-calculation/{calculationId}/data endpoint. Upon subscribing to a report, you will always receive an initial event with a calculationId through which you can access the current state of the report's data.
```

Is there any good place here to also specify the schema for the Events being sent?

---

Yes, i will add a schema for that.

---

@tomwwinter , do we have any place to put a more detailed "description", along the lines of what I drafted here in my initial comment?

---

added externalDoc link